components/postmark/postmark.app.mjs

```diff
@@ -1,11 +1,208 @@
+      type: "string",
+      label: "CC email address(es)",
+      description:
+        "Cc recipient email address. Multiple addresses are comma separated. Max 50.",
+      optional: true,
+    },
+    bcc_email: {
+      type: "string",
+      label: "BCC email address(es)",
+      description:
+        "Bcc recipient email address. Multiple addresses are comma separated. Max 50.",
+      optional: true,
+    },
+    tag: {
+      type: "string",
+      label: "Tag",
+      description:
+        "Email tag that allows you to categorize outgoing emails and get detailed statistics.",
+      optional: true,
+    },
+    reply_to: {
+      type: "string",
+      label: "\"Reply To\" email address",
+      description:
+        "Reply To override email address. Defaults to the Reply To set in the sender signature.",
+      optional: true,
+    },
+    custom_headers: {
+      type: "string[]",
+      label: "Custom Headers",
+      description: "List of custom headers to include.",
+      optional: true,
+    },
+    track_opens: {
+      type: "boolean",
+      label: "Track Opens",
+      description: "Activate open tracking for this email.",
+      optional: true,
+    },
+    track_links: {
+      type: "string",
+      label: "Track Links",
+      description:
+        "Activate link tracking for links in the HTML or Text bodies of this email.",
+      optional: true,
+      options: [
+        "None",
+        "HtmlAndText",
+        "HtmlOnly",
+        "TextOnly",
+      ],
+    },
+    attachments: {
+      type: "string[]",
+      label: "Attachments",
+      description: `Each attachment should be a string with the parameters separated by a pipe character \`|\`, in the format: \`Name|Content|ContentType\`. Alternatively, you can pass a string representing an object. All three parameters are required:
+      \\
+      \\
+      \`Name\` - the filename with extension, i.e. \`readme.txt\`
+      \\
+      \`Content\` - the base64-encoded string with the binary data for the file, i.e. \`dGVzdCBjb250ZW50\`
+      \\
+      \`ContentType\` - the MIME content type, i.e. \`text/plain\`
+      \\
+      \\
+      Example with pipe-separated parameters: \`readme.txt|dGVzdCBjb250ZW50|text/plain\`
+      \\
+      Example with JSON-stringified object: \`{"Name":"readme.txt","Content":"dGVzdCBjb250ZW50","ContentType":"text/plain"}\`
+      `,
+      optional: true,
+    },
+    metadata: {
+      type: "object",
+      label: "Metadata",
+      description: "Custom metadata key/value pairs.",
+      optional: true,
+    },
+    message_stream: {
+      type: "string",
+      label: "Message stream",
+      description:
+        "Set message stream ID that's used for sending. If not provided, message will default to the outbound transactional stream.",
+      optional: true,
+    },
+  },
   methods: {
     // this.$auth contains connected account data
     authKeys() {
       console.log(Object.keys(this.$auth));
     },
+    async listTemplates() {
```

Comment:
I would suggest trying to simplify this method, something like this:

```js
    async listTemplates($ = this) {
      const { data } = await axios($, {
        url: "https://api.postmarkapp.com/templates?count=500&offset=0",
        headers: this.getHeaders(),
        method: "GET",
      });

      if (data.TotalCount > 0) {
        return data.Templates
          .filter((obj) => obj.TemplateType === "Standard")
          .map((obj) => ({
            label: obj.Name,
            value: obj.Alias,
          }));
      }

      return [];
    },
```

Pipedream can handle if there are no options to be selected.

Also, if possible, [we prefer to use `await` instead of `promise chaining`](https://pipedream.com/docs/code/nodejs/async/#await-all-promises)

Reply:
And instead of requesting for all options in one API call, it would be better to paginate them like [this example](https://github.com/PipedreamHQ/pipedream/blob/master/components/google_drive/google_drive.app.mjs#L64) or refer to [here in the docs](https://pipedream.com/docs/components/guidelines/#async-options)

Reply:
Simplified the method considerably, and removed the promise chaining.

Great call on the pagination. Implemented that as well.